src/include/buffer.h

```diff
@@ -703,7 +703,9 @@ namespace buffer CEPH_BUFFER_API {
 
       void advance(int o) = delete;
       void advance(unsigned o);
+#if UINT_WIDTH != SIZE_WIDTH
```

Comment:
@rzarzynski why do we need this overload? can we simply remove it?

Reply:
In general we could remove it as the `buffer` library tends to disrespects `size_t` – `list::_len` and `ptr::_len` are both `unsigned`. There are some exceptions like `size_t buffer::ptr::get_offset()` (used altogether with `::advance` in `denc.h`) but they deserve rework as well.

Reply:
@rzarzynski but in this very case, we simply cast `size_t` to `unsigned`. so we could silently trim a 64bits integer to 32bit on 64bits architecture.

Reply:
@tchaikov: ah, I see what you mean. Yup, we could go that way. :+1:

Reply:
@rzarzynski thanks for confirming this!

@the78mole could you remove `advance(size_t o)` in a separated commit?

Reply:
posted PR at #28278 to remove this method.